polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -1464,8 +1464,8 @@ Value MLIRScanner::GetAddressOfBaseClass(
       if (!PT)
         Val = Builder.create<polygeist::Pointer2MemrefOp>(Loc, NT, Val);
       else {
-        if (Val.getType() != NT)
-          Val = Builder.create<LLVM::BitcastOp>(Loc, PT, Val);
+        if (Opt.getAddressSpace() != PT.getAddressSpace())
```

Comment:
Question: is it possible here to have a source pointer that has different element type and also different address space. Of course, if we only use opaque pointers the element type is not there. But with typed pointers, should we possibly do a bitcast to change the element type (in addition to the address space cast) ?

Reply:
With typed pointers, a sequence of bitcast *and* AS-cast might be necessary, yes (before the change, an illegal bitcast would have been created in that scenario). 

However, as typed pointer support is on the way out (in the last opaque pointer PR, we agreed to drop this functionality as soon as it's done on the `sycl` branch in the next weeks) and opaque is the default now, I decided to not support typed pointers here.